### PR TITLE
feat: add datasource type postgresql

### LINKED ISSUE #490: able to add a new datasource type PG
To support pg vector,we need a better way to conigure postgresql credentials inlcuding:
- url
- authsecret
- etc...

To keep consitent with minio, we can implement a new datasource `Postgresql`.

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes

Fixes #490

### CODE REVIEW

File: api/base/v1alpha1/datasource_types.go [hunk 1]
```diff
@@ -54,6 +54,37 @@ type OSS struct {
...
+// PostgreSQL defines info for PostgreSQL
+//
+// ref: https://github.com/jackc/pgx/blame/v5.5.1/pgconn/config.go#L409
+// they are common standard PostgreSQL environment variables
+// For convenience, we use the same name.
+//
+// The PGUSER/PGPASSWORD/PGPASSFILE/PGSSLPASSWORD parameters have been intentionally excluded
+// because they contain sensitive information and are stored in the secret pointed to by `endpoint.authSecret`.
+type PostgreSQL struct {
+	Host               string `json:"PGHOST,omitempty"`
```
Comment (bjwswang) [inline api/base/v1alpha1/datasource_types.go:66]: @Abirdcfly Can we reuse the content in `Endpoint`?

Comment (Abirdcfly) [reply api/base/v1alpha1/datasource_types.go:66]: We reuse endpoint, and here is just a list of all the other parameters that we can use.

Comment (bjwswang) [reply api/base/v1alpha1/datasource_types.go:66]: So how they (endpoit.url and Posgresql.Host ) work together ?

File: pkg/datasource/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	panic("implement me")
+}
+
+func (p *PostgreSQL) ListObjects(ctx context.Context, source string, info any) (any, error) {
+	// TODO implement me
+	panic("implement me")
+}
+
+// SetPGEnv will export all pg setting to environment variable
+func SetPGEnv(ctx context.Context, c client.Client, dc dynamic.Interface, config *v1alpha1.PostgreSQL, endpoint *v1alpha1.Endpoint) error {
```
Comment (bjwswang) [inline pkg/datasource/postgresql.go:91]: What if we have two postgresql datasources ?  Will they have conflicts in these environment variables?

File: api/base/v1alpha1/datasource_types.go (hunk 1 above)
Comment (Abirdcfly) [reply api/base/v1alpha1/datasource_types.go:66]: set `endpoint.url` to  `pgx.Connect(ctx, connString)`, and set PostgreSQL.Host to env `PGHOST` (other PostgreSQL.xxx to env PGxxx), pgx (or other pg go library) will merge settings like `mergeSettings(defaultSettings, envSettings, connStringSettings)` (https://github.com/jackc/pgx/blob/6f2ce92356f35263258ef7baa4cde86a3e1c6255/pgconn/config.go#L250)

File: pkg/datasource/postgresql.go (hunk 2 above)
Comment (Abirdcfly) [reply pkg/datasource/postgresql.go:91]: if we have two more postgresql datasources, we should not set env, i guess...

Comment (bjwswang) [reply pkg/datasource/postgresql.go:91]: well... then we need to find another way to pass the credentials and other configurations to pgx I think

Comment (Abirdcfly) [reply pkg/datasource/postgresql.go]: Some attempts later, I think it's still a good idea to configure environment variables, delete them after creating the pool, and ensure that multiple configurations don't affect each other when created by using `sync.Mutex` to lock it.

I realized that if I configure the advanced configurations directly, it's equivalent to reimplementing the https://github.com/jackc/pgx/blob/6f2ce92356f35263258ef7baa4cde86a3e1c6255/pgconn/config.go#L229 `ParseConfigWithOptions` function again.

File: controllers/datasource_controller.go [hunk 3]
```diff
@@ -154,23 +154,26 @@ func (r *DatasourceReconciler) Checkdatasource(ctx context.Context, logger logr.
...
-		}
 		ds, err = datasource.NewOSS(ctx, r.Client, nil, endpoint)
 		if err != nil {
 			return r.UpdateStatus(ctx, instance, err)
 		}
 		info = instance.Spec.OSS.DeepCopy()
 	case arcadiav1alpha1.DatasourceTypeRDMA:
 		return r.UpdateStatus(ctx, instance, nil)
+	case arcadiav1alpha1.DatasourceTypePostgreSQL:
+		_, err = datasource.NewPostgreSQL(ctx, r.Client, nil, instance.Spec.PostgreSQL, endpoint)
```
Comment (bjwswang) [inline controllers/datasource_controller.go:175]: we should keep consistent with oss  which returns a info and do a basic `Stat` check

File: pkg/datasource/postgresql.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+	}
+	if err := pool.Ping(ctx); err != nil {
+		return nil, err
+	}
+	return &PostgreSQL{pool}, nil
+}
+
+func (p *PostgreSQL) Stat(ctx context.Context, info any) error {
+	// TODO implement me
+	panic("implement me")
```
Comment (bjwswang) [inline pkg/datasource/postgresql.go:202]: Do a basic stat chekc or return nil directly ?

File: deploy/charts/arcadia/Chart.yaml [hunk 5]
```diff
@@ -15,6 +15,10 @@ sources:
 maintainers:
   - name: bjwswang
     url: https://github.com/bjwswang
+  - name: Abirdcfly
```
Comment (bjwswang) [inline deploy/charts/arcadia/Chart.yaml:18]: `nkwangleiGIT`

**Review by bjwswang — APPROVED** [review]

**Pull request merged**